analysis/src/CompletionBackEnd.ml

```diff
@@ -2153,16 +2156,20 @@ Note: The `@react.component` decorator requires the react-jsx config to be set i
                | (Labelled n | Optional n) when name = n -> true
                | _ -> false))
     in
+    let expandOption =
+      match targetLabel with
+      | None | Some ((Unlabelled _ | Labelled _), _) -> false
+      | Some (Optional _, _) -> true
+    in
     match targetLabel with
     | None -> []
-    | Some (Optional _, typ) ->
-      typ
-      |> completeTypedValue ~env ~envWhereCompletionStarted ~full ~prefix
-           ~expandOption:true ~includeLocalValues:true ~completionContext:None
-    | Some ((Unlabelled _ | Labelled _), typ) ->
-      typ
-      |> completeTypedValue ~env ~envWhereCompletionStarted ~full ~prefix
-           ~expandOption:false ~includeLocalValues:true ~completionContext:None)
+    | Some (_, typ) -> (
+      match typ |> resolveNested ~env ~package:full.package ~nested with
+      | None -> []
+      | Some (typ, env, completionContext) ->
+        typ
+        |> completeTypedValue ~env ~envWhereCompletionStarted ~full ~prefix
+             ~expandOption ~includeLocalValues:true ~completionContext))
```

Comment:
Not introduced here, but it might be simpler to use a function that expands the option before calling `completeTypedValue`, so less indirect action at a distance and recursive call to implement the "expand on only once" logic.

Reply:
Fixed